app/Models/PeriodicEvent.php

```diff
@@ -112,13 +121,19 @@ public function isExtended(): bool
         return $this->extended_end_date != null;
     }
 
+
     /**
      * Handle the start of the PeriodicEvent.
      */
     public function handleStart(): void
     {
-        //Get the corresponding controller and call its start method
-        app($this->event_model)->handlePeriodicEventStart();
+        switch ($this->event_name) {
+            case self::SEMESTER_EVALUATION_PERIOD:
+                event(new SemesterEvaluationPeriodStart($this));
+                break;
```

Comment:
This seems not to be open-closed; can we circumvent this somehow? Maybe through singletons derived from this class.